cmake/EthDependencies.cmake

```diff
@@ -31,14 +31,27 @@ endif()
 
 set(BOOST_COMPONENTS "filesystem;unit_test_framework;program_options;system")
 
+# CMake >= 3.30 should not use the vendored boost
+if(POLICY CMP0167)
+	cmake_policy(SET CMP0167 NEW)
+endif()
 if (WIN32)
 	# Boost 1.77 fixes a bug that causes crashes on Windows for some relative paths in --allow-paths.
 	# See https://github.com/boostorg/filesystem/issues/201
 	find_package(Boost 1.77.0 QUIET REQUIRED COMPONENTS ${BOOST_COMPONENTS})
 else()
 	# Boost 1.65 is the first to also provide boost::get for rvalue-references (#5787).
 	# Boost 1.67 moved container_hash into is own module.
-	find_package(Boost 1.67.0 QUIET REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+	# Boost 1.70 comes with its own BoostConfig.cmake and is the new (non-deprecated) behavior
+	find_package(Boost 1.70.0 QUIET COMPONENTS ${BOOST_COMPONENTS})
+	if (NOT ${Boost_FOUND})
+		# If the boost version is < 1.70.0, there is no boost config delivered with it, revert to old behavior
+		# todo drop this once cmake minimum version >= 3.30 is reached
+		if(POLICY CMP0167)
+			cmake_policy(SET CMP0167 OLD)
```

Comment:
Can't we just set this unconditionally here?

Reply:
Could be cmake < 3.30 in which case the policy doesn't exist yet